packages/app/src/components/Settings/Settings.js

```diff
@@ -0,0 +1,267 @@
+            
+            setNewGrantCycle({openedOn: "", closedOn: "", name: ""})
+            createButton.current.blur();
+            getGrantCycles();
+        }
+        catch(ex) {
+            if (ex.response && ex.response.status === 400) {
+                setErrors(ex.response.data);
+                createButton.current.blur();
+            }
+        }
+    
+    }
+    const handleUpdate = async (e) => {
+        
+        try{
+            console.log(gcToEdit);
+            const {data} = await grantCycleAPI.updateGrantCycle(gcToEdit);
+            
+            setGcToEdit({openedOn: "", closedOn: "", name: ""})
+            getGrantCycles();
+            setShowEditModal(false);
+        }
+        catch(ex) {
+            if (ex.response && ex.response.status === 400) {
+                setEditErrors(ex.response.data);
+            }
+        }
+    }
+
+    const handleEdit = gc => {
+        setGcToEdit({ openedOn: gc.openedOn.split("T")[0], closedOn: gc.closedOn.split("T")[0], name: gc.name, id: gc.id });
+        setShowEditModal(true);
+    }
+
+    useEffect(() => {
+        handleDisableButton(newGrantCycle);
+    }, [newGrantCycle])
+
+    useEffect(() => {
+        if (gcToEdit)
+            handleDisableEditButton(gcToEdit);
+    }, [gcToEdit])
+
+    const handleDisableButton = (gc) => {
+        if (gc.openedOn !== "" && gc.closedOn !== "") {
+            const millisecondsInADay = 86400000;
+            const startDate = new Date(gc.openedOn);
+            const endDate = new Date(gc.closedOn);
+            const result = endDate - startDate >= millisecondsInADay ? false : true;
+            
+            setDisableButton(result);
+            if (!result) {
+                setErrors("")
+                if (gc.name === "") {
+                    setDisableButton(true);
+                    setErrors("Please enter a name for the grant cycle");
+                }
+            }
+            else setErrors("Start Date must be earlier than End Date");
+        }
+        else {
+            setDisableButton(true);
+        }
+    }
+    
+    const handleDisableEditButton = (gc) => {
+        if (gc.openedOn !== "" && gc.closedOn !== "") {
+            const millisecondsInADay = 86400000;
+            const startDate = new Date(gc.openedOn);
+            const endDate = new Date(gc.closedOn);
+            const result = endDate - startDate >= millisecondsInADay ? false : true;
+            
+            setDisableEditButton(result);
+            if (!result) {
+                setEditErrors("")
+                if (gc.name === "") {
+                    setDisableEditButton(true);
+                    setEditErrors("Please enter a name for the grant cycle");
+
+                }
+                const closed = moment(gc.closedOn);
+                if (moment().isAfter(closed))
+                    setDisableEditButton(true);
+            }
+            else setEditErrors("Start Date must be earlier than End Date");
```

Comment:
Should probably have brackets around this else to be stylistically consistent with the rest of the code in this block.